### DISCUSSION TITLE
5900x (and similar) - 80 plots per day guide

### DISCUSSION THREAD

**Discussion by vladaepro** [madMAx43v3r/chia-plotter#917] [Show and tell]
I've seen that mane people with similar cpu and other hardware have a lot worse plot times than mine and struggling to get below 20 minutes, so i would like to share how i'm getting ~80 plots per day.

**1. HARDWARE**
 5900x, 128GB DDR4 3600 CL18, NVME 970 **pro** 512GB
Think that NVME play big role here - people have tendency to buy nvme with fastest declared speed, but that's only maximum theoretical speed while using cache. **Far more important than that is sustained write speed where 970 pro excels**. I tried also two 970 pro in raid, but gain is so minor that it's not worth.

**2. BIOS**
 there are two important tweaks in bios. First one is about cpu - PBO should be enabled (to advanced mode) and Curve optimizer se to -10 for all cores (if you are lucky, you can go maybe even -20 or -30, but i found -10 super stable on all of my plotters)
Second one is fclk (Infinity fabric clock). It's not enough just to load XMP to set ram to 3600, if you left fclk to auto, most motherboards will use 1:2 ratio - 900MHz. **Set this manually to 1800MHz**.
Important note - it wasn't possible on my Asrock x570 phantom gaming pro before bios version 4.20 which introduced AMD AM4 AGESA Combo V2 PI 1.2.0.3 Patch C. If you can't get it stable with fclk 1800MHz, set to auto but expect about 3 plots per day less.

**3. SOFTWARE**
Think that most important thing here is OS. I'm getting best results with **Clear Linux, kernel 5.12** (5.13 is slower, beware of that!). Maybe some other distros have similar performance but i didn't tried except Ubuntu which was way slower.
Make sure you are using XFS for temp drive, not ext4
110G tmpfs is assumed
Plotter arguments i'm using (beside defaults/assumed):  -r 12 -u 512 -K 2  

With all of above, my first plot is : Total plot creation time was **1062.74 sec (17.7123 min)**

**4. OTHER**
There can be quite big drop in speed after first plot (because of copying it somewhere), depends what you are using for that purpose. My advice is to avoid rsync (Plotman) if possible, as it takes cpu time because of ssl encryption, and also because of io wait while copying. If your HDDs are on the plotter, best way is to use second fast nvme as final destination, and some script to copy files to hdds (without using ssh/rsync), or find some plots manager which doesn't use ssl.

But fastest plots times i got is by using dedicated server with 10gbe network and fast nvme 'buffer'. On plotter, final drive is same as temp drive, so there's no copy when plot is finished, just rename. Then script checks for available plots and send to server by using nc (netcat). Server also have it's own script which distribute available plots to hdds. 
By using that way, plot times (except first one) are about this: Total plot creation time was **1077.5 sec (17.9583 min)**

Second approach (with server) make sense if you are 'hard core' plotter and have multiple PC-s, first one is simpler, but expect 2-3 plots less per day.

Hope this guide can help you getting similar times as mine. Feel free to suggest if you think it could be additionally improved.

**Comment by idog8818** [comment]
One of the biggest issue is when the plot time is less then 40mins, due to hard drive speed or 1G network limit, the hard drive will hit the limit. I didn't setup for raid 0, just single partition on another low power consuming pc. The drive will only handle around 60MB/s. which will need 40mins to complete copy 1 plot.

**Reply by vladaepro** [reply]
1G network is capable of about 100 MB/sec, and majority of HDDs today write with easy 100+ MB/s, which means that single plot is written for about 15-16 minutes, so no issues with plotting time of 17-18min. However, in my setup i'm using 10Gbe nework and server which can write to more drives simultaneously so i never have issue with transfer/write speed.

**Reply by idog8818** [reply]
Just realized I haven't enabled hard drive cache which lead to low performance. Enabled and now writing at 120MB/s

**Reply by hajes** [reply]
I do one plot in ~37min, final plot copy speed ~118MBs/15min over eth/Samba ;-)

Any modern HDD can handle at least 120MBs (the crap eco ones). Average speed is ~150MBs...the best, data center, drives can easily do 220-250MBs ;-)

**Comment by d4672531** [comment]
> I've seen that mane people with similar cpu and other hardware have a lot worse plot times than mine and struggling to get below 20 minutes, so i would like to share how i'm getting ~80 plots per day.
> 
> **1. HARDWARE** 5900x, 128GB DDR4 3600 CL18, NVME 970 **pro** 512GB Think that NVME play big role here - people have tendency to buy nvme with fastest declared speed, but that's only maximum theoretical speed while using cache. **Far more important than that is sustained write speed where 970 pro excels**. I tried also two 970 pro in raid, but gain is so minor that it's not worth.
> 
> **2. BIOS** there are two important tweaks in bios. First one is about cpu - PBO should be enabled (to advanced mode) and Curve optimizer se to -10 for all cores (if you are lucky, you can go maybe even -20 or -30, but i found -10 super stable on all of my plotters) Second one is fclk (Infinity fabric clock). It's not enough just to load XMP to set ram to 3600, if you left fclk to auto, most motherboards will use 1:2 ratio - 900MHz. **Set this manually to 1800MHz**. Important note - it wasn't possible on my Asrock x570 phantom gaming pro before bios version 4.20 which introduced AMD AM4 AGESA Combo V2 PI 1.2.0.3 Patch C. If you can't get it stable with fclk 1800MHz, set to auto but expect about 3 plots per day less.
> 
> **3. SOFTWARE** Think that most important thing here is OS. I'm getting best results with **Clear Linux, kernel 5.12** (5.13 is slower, beware of that!). Maybe some other distros have similar performance but i didn't tried except Ubuntu which was way slower. Make sure you are using XFS for temp drive, not ext4 110G tmpfs is assumed Plotter arguments i'm using (beside defaults/assumed): -r 12 -u 512 -K 2
> 
> With all of above, my first plot is : Total plot creation time was **1062.74 sec (17.7123 min)**
> 
> **4. OTHER** There can be quite big drop in speed after first plot (because of copying it somewhere), depends what you are using for that purpose. My advice is to avoid rsync (Plotman) if possible, as it takes cpu time because of ssl encryption, and also because of io wait while copying. If your HDDs are on the plotter, best way is to use second fast nvme as final destination, and some script to copy files to hdds (without using ssh/rsync), or find some plots manager which doesn't use ssl.
> 
> But fastest plots times i got is by using dedicated server with 10gbe network and fast nvme 'buffer'. On plotter, final drive is same as temp drive, so there's no copy when plot is finished, just rename. Then script checks for available plots and send to server by using nc (netcat). Server also have it's own script which distribute available plots to hdds. By using that way, plot times (except first one) are about this: Total plot creation time was **1077.5 sec (17.9583 min)**
> 
> Second approach (with server) make sense if you are 'hard core' plotter and have multiple PC-s, first one is simpler, but expect 2-3 plots less per day.
> 
> Hope this guide can help you getting similar times as mine. Feel free to suggest if you think it could be additionally improved.

Do you install clear Linux inside or outside Windows? Does this have any impact on performance? 

Do you mean we need to use one PC for plotting and another PC for farming? Because using one PC for both will affect the performance for plotting?

Thanks

**Reply by vladaepro** [reply]
it's dedicated native linux installed. i was talking here how to get max out of specific hardware, you can use it with windows but then you can't quite match the performance.

**Comment by brianwfreeman** [comment]
Thanks for sharing your settings.  Question re: Clear Linux -- are the install instructions the same as Ubuntu to install MadMax on Clear?

**Reply by vladaepro** [reply]
yes, it's quite straightforward, like for other distros

**Reply by brianwfreeman** [reply]
Great to know - I have only tried Ubuntu and Linux Mint - I am very interested to try a new distro especially if it gives performance improvements.

**Reply by brianwfreeman** [reply]
And sorry to bombard you with questions about this, but how did you discover Clear and know that it would be faster?  I just did a Google search for "fastest Linux distributions" and the results are basically websites talking about what runs fastest on old computers.

**Reply by idog8818** [reply]
> 
> 
> And sorry to bombard you with questions about this, but how did you discover Clear and know that it would be faster? I just did a Google search for "fastest Linux distributions" and the results are basically websites talking about what runs fastest on old computers.

I installed windows 10 at early stage, 40 mins per plot, I then tried linux and only 22mins per plot. completed same machine.

**Reply by vladaepro** [reply]
> And sorry to bombard you with questions about this, but how did you discover Clear and know that it would be faster? I just did a Google search for "fastest Linux distributions" and the results are basically websites talking about what runs fastest on old computers.

well, i just heard about it from some people at early stage of madmax plotter, and i compared to ubuntu which i had installed. I'm not sure which aspect is most important for speed difference, but somehow think it's file system handling, it seems that XFS is better optimized for heavy parallel IO than on some other distros.

**Comment by hajes** [comment]
Will try ClearLinux because Windows is crap with RAMdisk.

MadMax + Ryzen 3950X + Corsair MP600 2TB = <35min
same setup, but 111GB RAMdisk via ImDisk is ~37min ROFL

In synth tests RAM disk does over 3000MB/s and MP600 manages around 1200-1500MBs sustainable. It makes you think...

**Comment by ruixingchen** [comment]
my 5950X is at 24 min, the 2666 CL19 RAM causes this I think
